### PR TITLE
Updates Hub CR to prevent loss of data on operator upgradation

### PR DESCRIPTION
**Pull request by PuneetPunamiya**
When operator is upgraded, new hub db is getting created and all the
data from the db gets deleted

Hence this patch fixes this by applying the pvc first by injecting the
owner reference and then filtering out pvc from the installerset

Signed-off-by: Puneet Punamiya <ppunamiy@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

**Comment by nikhil-thomas** [comment]
the logic behind the fix is good. Could we move the added code in this patch into a separate function.

sturctural suggestions.
1. create a function or block which reads the manifests
2. do the filtering of resources outside `setupAndCreateInstallerSet`
3. make `setupAndCreateInstallerSet` filtered manifests as input instead of the manifest filepath.

File: pkg/reconciler/kubernetes/tektonhub/tektonhub.go [hunk 1]
```diff
@@ -192,10 +193,22 @@ func (r *Reconciler) manageUiComponent(ctx context.Context, th *v1alpha1.TektonH
...
 		th.Status.MarkUiInstallerSetNotAvailable("UI installer set not available")
 		uiLocation := filepath.Join(hubDir, "ui")
-		err := r.setupAndCreateInstallerSet(ctx, uiLocation, th, uiInstallerSet, version, ui)
+
+		manifest, err := r.getManifest(ctx, th, uiLocation)
+		if err != nil {
+			return err
+		}
+
+		err = applyPVC(ctx, manifest, th)
```
Comment (sm43) [inline pkg/reconciler/kubernetes/tektonhub/tektonhub.go:202]: does UI has a pvc?

File: pkg/reconciler/kubernetes/tektonhub/tektonhub.go [hunk 2]
```diff
@@ -248,10 +267,17 @@ func (r *Reconciler) manageDbMigrationComponent(ctx context.Context, th *v1alpha
...
+		manifest, err := r.getManifest(ctx, th, dbMigrationLocation)
+		if err != nil {
+			return err
+		}
+
+		err = r.setUpAndCreateInstallerSet(ctx, *manifest, th, dbMigrationInstallerSet, version, dbMigration)
 		if err != nil {
 			return err
 		}
+
```
Comment (sm43) [inline pkg/reconciler/kubernetes/tektonhub/tektonhub.go:280]:
```suggestion
```

**Pull request merged**